runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -1930,13 +1930,624 @@ TR::Register *J9::Power::TreeEvaluator::anewArrayEvaluator(TR::Node *node, TR::C
       return TR::TreeEvaluator::VMnewEvaluator(node, cg);
    }
 
+// put source*imm in target as long as imm is a non-negative power of two
+static void generateImmMultiplicationWithShift(TR::Node *node, TR::CodeGenerator *cg,
+                                                        TR::Register *targetReg,
+                                                        TR::Register *sourceReg,
+                                                        int32_t imm)
+   {
+   TR::Compilation *comp = cg->comp();
+   TR_ASSERT_FATAL(comp->target().is64Bit(),
+      "immMultiplicatoinWithShift is only supported on 64-bit JVMs!");
+
```

Comment:
you are assuming sourceReg is an unsigned 32bit value, otherwise this isn't entirely correct for every scenario.